engine/src/render/cascaded_shadow_maps_rasterizer/plugin.cpp

```diff
@@ -0,0 +1,314 @@
+#include <cubos/core/io/window.hpp>
+#include <cubos/core/reflection/external/primitives.hpp>
+
+#include <cubos/engine/assets/plugin.hpp>
+#include <cubos/engine/render/camera/camera.hpp>
+#include <cubos/engine/render/camera/draws_to.hpp>
+#include <cubos/engine/render/camera/orthographic.hpp>
+#include <cubos/engine/render/camera/perspective.hpp>
+#include <cubos/engine/render/camera/plugin.hpp>
+#include <cubos/engine/render/cascaded_shadow_maps/plugin.hpp>
+#include <cubos/engine/render/cascaded_shadow_maps_rasterizer/plugin.hpp>
+#include <cubos/engine/render/g_buffer/g_buffer.hpp>
+#include <cubos/engine/render/g_buffer/plugin.hpp>
+#include <cubos/engine/render/lights/directional.hpp>
+#include <cubos/engine/render/lights/plugin.hpp>
+#include <cubos/engine/render/mesh/mesh.hpp>
+#include <cubos/engine/render/mesh/plugin.hpp>
+#include <cubos/engine/render/mesh/pool.hpp>
+#include <cubos/engine/render/mesh/vertex.hpp>
+#include <cubos/engine/render/shader/plugin.hpp>
+#include <cubos/engine/render/shadows/directional_caster.hpp>
+#include <cubos/engine/render/shadows/plugin.hpp>
+#include <cubos/engine/render/voxels/grid.hpp>
+#include <cubos/engine/render/voxels/plugin.hpp>
+#include <cubos/engine/transform/local_to_world.hpp>
+#include <cubos/engine/transform/plugin.hpp>
+#include <cubos/engine/window/plugin.hpp>
+
+using namespace cubos::core::gl;
+using cubos::core::io::Window;
+using cubos::engine::RenderMeshVertex;
+
+namespace
+{
+    // Holds the data sent per scene to the GPU.
+    struct PerScene
+    {
+        glm::mat4 lightViewProj[cubos::engine::DirectionalShadowCaster::MaxSplits];
+        int numSplits;
```

Comment:
I think you might need to include padding here (`int padding[3];`), otherwise scary UB might happen, due to the struct on the GPU having 12 bytes of padding.